pyresample/geometry.py

```diff
@@ -391,6 +391,28 @@ def get_edge_bbox_in_projection_coordinates(self, frequency: Optional[int] = Non
         x, y = self._get_bbox_elements(self.get_proj_coords, frequency)
         return np.hstack(x), np.hstack(y)
 
+    def boundary(self, frequency=None, force_clockwise=False):
+        """Retrieve the AreaBoundary object.
+
+        Parameters
+        ----------
+        frequency:
+            The number of points to provide for each side. By default (None)
+            the full width and height will be provided.
```

Comment:
Technically, at least in the old method, this was an integer equivalent to a "stride" in slicing. At least I think it was. So you would say "I want a boundary made up of every N pixels along the edge".

Reply:
Yes this was the old behavior. But in April-May, I noted to @mraspaud that such striding behavior was causing to often drop out large portions of swath/area between the last point selected and the actual corner of the swath/area. 
In the context of the gradient search PR, the behavior has been changed. See `_get_bbox_slices` for more details: https://github.com/pytroll/pyresample/blob/1268d904c9bcd01342173dcee73809da59cc79c1/pyresample/geometry.py#L332

What I discussed with Martin in these days, in one of the upcoming PRs, is to allow specifying `frequency` also as a tuple (n_x, n_y) to customize the number of points to select on the x and y sides.
This would help for example to speed up spherical operations for swath definitions where for the y-axis (cross-track scan direction), much fewer points (i.e only 2) are needed to get the accurate swath boundary.

I agree that the term "frequency" is misleading, but it is used in a large portion of the code and changing it I think would cause lot of backward incompatibilities.